journal-parsing/src/main/java/com/fnz/db2/journal/retrieve/ParameterListBuilder.java

```diff
@@ -0,0 +1,166 @@
+package com.fnz.db2.journal.retrieve;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+import com.fnz.db2.journal.retrieve.RetrievalCriteria.JournalCode;
+import com.fnz.db2.journal.retrieve.RetrievalCriteria.JournalEntryType;
+import com.ibm.as400.access.AS400Bin4;
+import com.ibm.as400.access.AS400Structure;
+import com.ibm.as400.access.AS400Text;
+import com.ibm.as400.access.ProgramParameter;
+
+// TODO remove now we've sanitised RetrievalCriteria
```

Comment:
I know that this comment was relocated from the RetrieveJournal file. But it suggests this code isn't necessary - do we still need it, or would that need more development?

Reply:
I'm not sure if we need it but now I've added diagnostics via the toString I'm more tempted to keep it.